src/parser/rogue/outlaw/modules/talents/Alacrity.js

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import Analyzer from 'parser/core/Analyzer';
+import SPELLS from 'common/SPELLS';
+import SpellIcon from 'common/SpellIcon';
+import { formatDuration, formatPercentage } from 'common/format';
+import StatisticBox from 'interface/others/StatisticBox';
+import AlacrityTimesByStacks from 'parser/rogue/outlaw/modules/talents/AlacrityTimesByStacks';
+
+const HASTE_PER_STACK = 0.02;
+
+class Alacrity extends Analyzer {
+
+  static dependencies = {
+    alacrityTimesByStacks: AlacrityTimesByStacks,
+  };
+
+  constructor(...args) {
+    super(...args);
+    this.active = this.selectedCombatant.hasTalent(SPELLS.ALACRITY_TALENT.id);
+  }
+
+
+  get alacrityTimesByStack() {
+    return this.alacrityTimesByStacks.alacrityTimesByStacks;
+  }
+
+
+  get uptime() {
+    return this.selectedCombatant.getBuffUptime(SPELLS.ALACRITY_BUFF.id) / this.owner.fightDuration;
+  }
+
+  get averageHaste() {
+    return this.alacrityTimesByStacks.averageAlacrityStacks*HASTE_PER_STACK;
+  }
+
+  statistic() {
+
+    return (
+      <StatisticBox
```

Comment:
I honestly don't know how often dropping of would happen in actual play.
From what I've seen you usually don't drop it off when having proper uptime on the boss if there are no errors in rotation.

My guess was that displaying number of times buff was dropped would mostly just turn up false positives but I might be completely wrong with that.

I would completely follow your call with that.

Reply:
You are right. It will always drop on fights like Vectis, but we cant suggest to swap the talent since alternatives are not viable.

In that case I will just suggest making it look like the weapon enchants and reuse at least some of the code.